internal/profiler/profiler.go

```diff
@@ -0,0 +1,36 @@
+package profiler
+
+import (
+	"os"
+
+	"cloud.google.com/go/profiler"
+	ddprofiler "gopkg.in/DataDog/dd-trace-go.v1/profiler"
+)
+
+// Init starts the supported profilers IFF the environment variable is set.
+func Init(svcName, version string, blockProfileRate int) error {
+	if os.Getenv("DD_ENV") != "" {
+		profileTypes := []ddprofiler.ProfileType{ddprofiler.CPUProfile, ddprofiler.HeapProfile}
+		// additional profilers have a performance impact and should be enabled with care
+		if os.Getenv("DD_PROFILE_ALL") != "" || blockProfileRate > 0 {
+			profileTypes = append(profileTypes, ddprofiler.MutexProfile, ddprofiler.BlockProfile)
+		}
+		return ddprofiler.Start(
```

Comment:
you are doing an early return here, and then won't start google cloud profiler.

Reply:
I think we want only one profiler enabled at a time

Reply:
Why? If that is what an operator wants, then they can only enable one envvar?